### PR TITLE
docs: Weyland-Yutani parser tutorial

### PR DESCRIPTION
**Pull request by gvwilson**
Long-overdue tutorial for Weyland-Yutani parser example.

### CODE REVIEW

**Comment by gvwilson** [comment]
ping @mdlu

File: docs/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+and then a checksum of the readings.
+Unfortunately,
+the documentation available to us does not specify how this checksum is calculated.
+
+## Defining Files
+
+After forking the [`allotropy` repository`][allotropy_repo]
+and creating a new branch for our work,
+we make a new directory `./src/allotropy/parsers/example_weyland_yutani/` for our work
+and create three files in that directory:
```
Comment (mdlu) [inline docs/tutorial.md:104]: this sentence seems to have been a bit misformatted, would you be able to fix this? 

image

File: docs/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+Our tests then compare the generated JSON against the saved files.
+More complex parsers may require tests against helper methods,
+and will probably have many more input-output pairs to check.
+
+## Exercises
+
+To check your understanding,
+look for the keyword `TODO` in the Weyland-Yutani example's source code
+and fill in the missing methods.
+You can submit your work as a pull request for review,
```
Comment (mdlu) [inline docs/tutorial.md:307]: were you planning to look at any pull requests that came in? I wouldn't want to place that burden on you, but I don't anticipate that we at Benchling would plan to look at these. as such, I would maybe frame the `TODOs` as more like thought exercises or extensions? something like:

> We've left the keyword `TODO` throughout the Weyland-Yutani example's source code in a handful of missing methods. These serve as thought exercises and examples of potential extensions to the base parser when handling more-complex files.

**Review by mdlu — CHANGES_REQUESTED** [review]
thank you so much for this, the tutorial is really comprehensive and looks really great! just leaving a couple of minor comments

File: docs/tutorial.md (hunk 1 above)
Comment (gvwilson) [reply docs/tutorial.md:104]: fixed - thank you

File: docs/tutorial.md (hunk 2 above)
Comment (gvwilson) [reply docs/tutorial.md]: updated - thank you

**Comment by gvwilson** [comment]
@mdlu thanks for the feedback - I think this one is ready to merge

**Review by mdlu — APPROVED** [review]

**Review by mdlu — APPROVED** [review]

**Pull request merged**